pkg/controller/apicast/apicast_logic_reconciler.go

```diff
@@ -418,40 +421,34 @@ func (r *APIcastLogicReconciler) reconcileDeployment(desiredDeployment appsv1.De
 	err := r.Client().Get(context.TODO(), r.namespacedName(&desiredDeployment), &existingDeployment)
 	if err != nil {
 		if errors.IsNotFound(err) {
+			r.Logger().Info(fmt.Sprintf("Creating %s", k8sutils.ObjectInfo(&desiredDeployment)))
 			err = r.Client().Create(context.TODO(), &desiredDeployment)
-			r.Logger().Info("Creating Deployment...")
 			return err
 		}
 		return err
 	}
 
 	changed := false
 
-	if !reflect.DeepEqual(existingDeployment.Spec.Replicas, desiredDeployment.Spec.Replicas) {
+	if existingDeployment.Spec.Replicas != desiredDeployment.Spec.Replicas {
 		existingDeployment.Spec.Replicas = desiredDeployment.Spec.Replicas
 		changed = true
 	}
-	if !reflect.DeepEqual(existingDeployment.Spec.Template.Spec.Containers[0].Image, desiredDeployment.Spec.Template.Spec.Containers[0].Image) {
+	if existingDeployment.Spec.Template.Spec.Containers[0].Image != desiredDeployment.Spec.Template.Spec.Containers[0].Image {
 		existingDeployment.Spec.Template.Spec.Containers[0].Image = desiredDeployment.Spec.Template.Spec.Containers[0].Image
 		changed = true
 
 	}
-	if !reflect.DeepEqual(existingDeployment.Spec.Template.Spec.ServiceAccountName, desiredDeployment.Spec.Template.Spec.ServiceAccountName) {
+	if existingDeployment.Spec.Template.Spec.ServiceAccountName != desiredDeployment.Spec.Template.Spec.ServiceAccountName {
 		changed = true
 		existingDeployment.Spec.Template.Spec.ServiceAccountName = desiredDeployment.Spec.Template.Spec.ServiceAccountName
 	}
 
-	if !reflect.DeepEqual(existingDeployment.Spec.Template.Spec.Containers[0].Env, desiredDeployment.Spec.Template.Spec.Containers[0].Env) {
-		changed = true
-		existingDeployment.Spec.Template.Spec.Containers[0].Env = desiredDeployment.Spec.Template.Spec.Containers[0].Env
-	}
+	updatedTmp := ReconcileEnvVar(&existingDeployment.Spec.Template.Spec.Containers[0].Env, desiredDeployment.Spec.Template.Spec.Containers[0].Env)
```

Comment:
The same concern/doubt appears here as the comment I've left below.
This has even bigger impact as this ends up as environment variables directly in the deployment.

Lets say you were to change from EmbeddedConfiguration config source to AdminPortal config source in the CR.
If I'm not wrong this would leave the EmbeddedConfiguration corresponding environment variable in the DeploymentConfig and also add the new one for AdminPortal. As EmbeddedConfiguration envvar has precedence over the AdminPortal one the apicast deployment would not behave as desired and would it be a bug?

Reply:
totally right, I am going to refactor this code